src/formatDate.js

```diff
@@ -7,8 +7,62 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(dateStr, fromFormat, toFormat) {
+  const formatMap = {
+    YYYY: (d) => d,
+    YY: (d) => d.slice(-2),
+    MM: (d) => d.padStart(2, '0'),
+    DD: (d) => d.padStart(2, '0'),
+  };
+
+  const parseDate = (date, format) => {
+    const sep = format.find((f) => !formatMap[f]);
+    const parts = date.split(sep);
+    const dateObj = {};
+
+    format.forEach((part, index) => {
+      if (formatMap[part]) {
+        dateObj[part] = parts[index];
+      }
+    });
+
+    return dateObj;
+  };
+
+  const convertYear = (year, from, to) => {
+    if (from === 'YYYY' && to === 'YY') {
+      return year.slice(-2);
+    } else if (from === 'YY' && to === 'YYYY') {
+      return year < 30 ? `20${year}` : `19${year}`;
```

Comment:
There is a logical error in the condition for converting from 'YY' to 'YYYY'. The comparison 'year < 30' should be 'parseInt(year, 10) < 30' to correctly compare the numeric value of the year.